scripts/wrappers/version.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+
+import click
+import json
+from common.utils import snap_data
+from common.utils import run
+
+
+VERSIONS_FILE_NAME = "versions.json"
+VERSIONS_FILE_PATH = snap_data() / VERSIONS_FILE_NAME
+
+
+VERSIONS = None
+
+
+def get_upstream_version(upstream: str) -> str:
+    global VERSIONS
+
+    if VERSIONS is None:
+        # Cache versions
+        VERSIONS = _read_versions_file()
+
+    return VERSIONS[upstream]
+
+
+def _read_versions_file():
+    with open(VERSIONS_FILE_PATH, mode="r") as versions_file:
+        versions = json.loads(versions_file.read())
+        return versions
+
+
+def get_snap_version_data():
+    output = run("snap", "list", "microk8s")
```

Comment:
I think this is not going to work well with strict confinement. For the revision number, probably best to use the `SNAP_REVISION` environment variable. Also, the snap version is set to `$KUBE_VERSION` (see https://github.com/ubuntu/microk8s/blob/1e538046bac00df1b5ff5a1bc95ac11011d96480/snap/snapcraft.yaml#L608), which we already have